File: hw/camacho/u2/Hw29ZooApp/ZooApp/src/main/java/ec/edu/espe/zooapp/model/Dog.java
package ec.edu.espe.zooapp.model;

/**
 *
 * @author mateo
 */
public class Dog extends Animal { 
    private String breed;
    private boolean isTrained;

    public Dog(String name, int age, String species, String breed, boolean isTrained) {
        super(name, age, species);
        this.breed = breed;
        this.isTrained = isTrained;
    }

    public String getBreed() {
        return breed;
    }

    public void setBreed(String breed) {
        this.breed = breed;
    }

    public boolean isTrained() {
        return isTrained;
    }

    public void setTrained(boolean isTrained) {
        this.isTrained = isTrained;
    }
}

File: ws/camacho/u3/JavaScriptPet/model/Dog.js
import Animal from './Animal.js';

class Dog {
    constructor(name, weight) {
        return new Animal({
            name,
            weight,
            type: 'dog',
            foodAmount: Dog.calculateFood(weight)
        });
    }

    static calculateFood(weight) {
        return weight * 0.04; 
    }
}

export default Dog;